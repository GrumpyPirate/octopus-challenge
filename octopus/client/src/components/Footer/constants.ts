import styled from 'styled-components';

import rem from '../../utils/styles';

export const Wrapper = styled.footer`
  padding: ${rem(20)} 0 ${rem(24)};
  color: ${({ theme }) => theme.palette.blueLight02};
  font-size: ${rem(12)};
  background-color: ${({ theme }) => theme.palette.blueDark02};
`;

export const Copy = styled.p`
  margin: 0;
`;
