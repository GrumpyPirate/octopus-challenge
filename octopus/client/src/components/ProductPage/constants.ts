import styled, { keyframes } from 'styled-components';

import rem from '../../utils/styles';

const spinner = keyframes`
  from {
    transform: rotateZ(0turn);
  }

  to {
    transform: rotateZ(1turn);
  }
`;

export const Loading = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column-reverse;
  min-height: ${rem(320)};
  color: ${({ theme }) => theme.palette.pink};

  svg {
    display: block;
    width: ${rem(48)};
    height: ${rem(48)};
    fill: currentColor;
    animation: ${spinner} 2s linear 0s infinite;
  }
`;

export const LoadingText = styled.h1`
  font-size: ${rem(16)};
  font-weight: 400;
  color: ${({ theme }) => theme.palette.bodyColor};
`;
