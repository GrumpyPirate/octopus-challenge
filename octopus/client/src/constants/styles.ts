import { css, DefaultTheme } from 'styled-components';
import { colord } from 'colord';

declare module 'styled-components' {
  export interface DefaultTheme {
    layout: {
      gutters: {
        small: string;
        large: string;
      };
    };
    palette: {
      blueDark01: string;
      blueDark02: string;
      blueMid01: string;
      blueMid02: string;
      blueLight01: string;
      blueLight02: string;
      pink: string;
      pinkDark: string;
      bodyBackground: string;
      bodyColor: string;
    };
  }
}

export const theme: DefaultTheme = {
  layout: {
    gutters: {
      small: '1rem', // 16w
      large: '2.375rem', // 38w
    },
  },
  palette: {
    blueDark01: '#070c1f',
    blueDark02: '#02122d',
    blueMid01: '#29457a',
    blueMid02: '#151e33',
    blueLight01: '#5280d6',
    blueLight02: '#7c90b4',
    pink: '#f60068',
    pinkDark: colord('#f60068').darken(0.125).toHex(),
    bodyBackground: '#070c1f',
    bodyColor: '#fff',
  },
};

export const screenSizes = {
  sm: '30em', // 480w
  md: '48em', // 768w
  lg: '64em', // 1024w
  xl: '80em', // 1280w
  xxl: '90em', // 1440w
};

export const buttonReset = css`
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;

  /* inherit font & color from ancestor */
  color: inherit;
  font: inherit;

  /* Normalize 'line-height'. Cannot be changed from 'normal' in Firefox 4+. */
  line-height: normal;

  /* Corrects font smoothing for webkit */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  /* Corrects inability to style clickable 'input' types in iOS */
  -webkit-appearance: none;

  /* Remove excess padding and border in Firefox 4+ */
  &::-moz-focus-inner {
    border: 0;
    padding: 0;
  }

  touch-action: manipulation;
`;
