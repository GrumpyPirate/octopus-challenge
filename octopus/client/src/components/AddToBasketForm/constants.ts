import styled from 'styled-components';

import { buttonReset } from '../../constants/styles';
import rem from '../../utils/styles';

export const Form = styled.form`
  padding: ${rem(10)} 0 ${rem(24)};
`;

export const TopRow = styled.div`
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  margin-bottom: ${rem(20)};
`;

export const Price = styled.h3`
  font-size: ${rem(40)};
  font-weight: 700;
  line-height: 1;
  white-space: nowrap;
  margin: 0;
`;

export const Pounds = styled.span`
  display: inline-block;
  vertical-align: top;
`;

export const Pence = styled.span`
  position: relative;
  top: 0.0625em;
  display: inline-block;
  vertical-align: top;
  font-size: 0.75em;
`;

export const SubmitButton = styled.button`
  ${buttonReset}
  display: block;
  width: 100%;
  padding: ${rem(12)} ${rem(24)};
  font-size: ${rem(24)};
  font-weight: 700;
  text-align: center;
  border-radius: ${rem(4)};
  background-color: ${({ theme }) => theme.palette.pink};
  position: relative;

  &::after {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    height: 50%;
    pointer-events: none;
    background-image: linear-gradient(
      to bottom,
      rgba(255, 255, 255, 0.5),
      rgba(255, 255, 255, 0.25)
    );
    border-top-right-radius: inherit;
    border-top-left-radius: inherit;
  }

  &:active {
    background-color: ${({ theme }) => theme.palette.pinkDark};
  }
`;
