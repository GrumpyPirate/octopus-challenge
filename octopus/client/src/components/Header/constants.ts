import styled from 'styled-components';

import { buttonReset } from '../../constants/styles';
import rem from '../../utils/styles';

export const Wrapper = styled.header`
  position: sticky;
  top: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: ${rem(12)} ${({ theme }) => theme.layout.gutters.small};
  background-color: ${({ theme }) => theme.palette.bodyBackground};
  z-index: 1;
`;

export const Logo = styled.img`
  display: block;
  width: ${rem(192)};
  height: ${rem(28)};
`;

export const Basket = styled.div`
  position: relative;
`;

export const BasketButton = styled.button`
  ${buttonReset}

  img {
    display: block;
    width: ${rem(30)};
    height: ${rem(30)};
  }
`;

export const BasketProductCount = styled.span`
  display: inline-block;
  padding: ${rem(2)} ${rem(6)};
  background-color: ${({ theme }) => theme.palette.pink};
  color: #fff;
  font-size: ${rem(12)};
  font-weight: 700;
  border-radius: ${rem(8)};
  position: absolute;
  top: 100%;
  right: 100%;
  transform: translate(calc(50% + ${rem(6)}), calc(-50% - ${rem(6)}));
`;
