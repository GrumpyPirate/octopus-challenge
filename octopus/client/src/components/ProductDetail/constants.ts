import styled, { css } from 'styled-components';

import { screenSizes } from '../../constants/styles';
import rem from '../../utils/styles';

export const ProductHeader = styled.header`
  @media (min-width: ${screenSizes.md}) {
    display: grid;
    grid-template-columns: ${rem(256)} 1fr;
    grid-gap: ${({ theme }) => theme.layout.gutters.large};
    padding: 0 ${({ theme }) => theme.layout.gutters.large};
  }

  @media (min-width: ${screenSizes.lg}) {
    grid-template-columns: ${rem(320)} 1fr;
  }
`;

export const ProductImage = styled.figure`
  position: relative;
  margin: 0 ${({ theme }) => theme.layout.gutters.large};
  height: ${rem(390)};

  img {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    width: 100%;
    height: 100%;
    object-fit: contain;
    object-position: center;
  }

  @media (min-width: ${screenSizes.md}) {
    margin: 0;

    img {
      object-position: top center;
    }
  }
`;

export const ProductHeading = styled.div`
  padding: ${rem(48)} ${({ theme }) => theme.layout.gutters.large} ${rem(24)};
  background-color: ${({ theme }) => theme.palette.blueDark02};

  @media (min-width: ${screenSizes.md}) {
    background: none;
    padding-top: 0;
    padding-right: 0;
    padding-left: 0;
  }
`;

export const ProductForm = styled.div`
  padding: 0 ${({ theme }) => theme.layout.gutters.large};

  @media (min-width: ${screenSizes.md}) {
    padding: 0;
    max-width: ${rem(400)};
  }
`;

export const ProductMain = styled.div`
  margin-top: ${rem(-32)};

  @media (min-width: ${screenSizes.md}) {
    margin-top: 0;
  }
`;

export const ProductName = styled.h1`
  font-size: ${rem(36)};
  font-weight: 700;
  line-height: 1.15;
  margin: 0 0 ${rem(16)};
`;

export const Subtitle = styled.h2`
  color: ${({ theme }) => theme.palette.blueLight01};
  font-size: ${rem(18)};
  font-weight: 400;
  margin: 0;
`;

export const Section = styled.section<{ lighter?: boolean }>`
  padding: ${rem(32)} 0 ${rem(28)};

  ${({ lighter }) =>
    !!lighter &&
    css`
      background-color: ${({ theme }) => theme.palette.blueDark02};
    `}
`;

export const SectionHeading = styled.h3`
  font-size: ${rem(28)};
  font-weight: 700;
  margin: 0 0 ${rem(16)};
`;

export const Description = styled.p`
  margin: 0;
`;

export const SpecificationList = styled.dl`
  display: grid;
  grid-gap: ${rem(16)};
  grid-template-columns: ${rem(176)} 1fr;
  margin: 0;
`;

export const SpecificationValue = styled.dd`
  margin: 0;
`;
