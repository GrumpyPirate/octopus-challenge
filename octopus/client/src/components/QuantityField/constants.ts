import styled from 'styled-components';
import { buttonReset } from '../../constants/styles';
import rem from '../../utils/styles';

export const Label = styled.label`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Quantity = styled.span`
  font-size: ${rem(32)};
  line-height: ${rem(40)};
  font-weight: 700;
  padding: 0 ${rem(8)};
`;

export const LabelText = styled.div`
  font-size: ${rem(14)};
  color: ${({ theme }) => theme.palette.blueLight01};
  text-transform: uppercase;
  margin-bottom: ${rem(4)};
`;

export const Button = styled.button`
  ${buttonReset}
  font-size: ${rem(24)};
  line-height: 1;
  width: ${rem(40)};
  height: ${rem(40)};
  border-radius: ${rem(4)};
  background-color: ${({ theme }) => theme.palette.blueLight01};

  &[disabled] {
    background-color: ${({ theme }) => theme.palette.blueMid02};
  }
`;

export const Controls = styled.div`
  display: flex;
  align-items: stretch;
`;
