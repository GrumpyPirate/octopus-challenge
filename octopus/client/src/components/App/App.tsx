import { useEffect } from 'react';
import { Route, Routes, useNavigate } from 'react-router-dom';

import Footer from '../Footer/Footer';
import Header from '../Header/Header';
import ProductPage from '../ProductPage/ProductPage';

const App = function () {
  const navigate = useNavigate();

  /**
   * Use a side-effect to:
   *  - Redirect the user onto the sole product page we have available, for the sake of this exercise.
   *    This is simply to demonstrate how we'd pull the product ID from the route params.
   */
  useEffect(() => {
    navigate('/products/1');
  }, [navigate]);

  return (
    <>
      <Header />
      <Routes>
        <Route path="/products/:productId" element={<ProductPage />} />
      </Routes>
      <Footer />
    </>
  );
};

export default App;
