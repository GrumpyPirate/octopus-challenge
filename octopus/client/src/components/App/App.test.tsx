import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import READ_PRODUCT_BY_ID from '../../api/queries';
import { theme } from '../../constants/styles';
import { BasketContextProvider } from '../../context/Basket/Basket';
import productFixture from '../../testing/fixtures/Product';
import App from './App';

const mockProductId = '1';
const mocks: MockedResponse[] = [
  {
    request: {
      query: READ_PRODUCT_BY_ID,
      variables: { id: mockProductId },
    },
    result: {
      data: { product: productFixture },
    },
  },
];

const withWrapper = function (element: ReactElement) {
  return (
    <MockedProvider mocks={mocks}>
      <ThemeProvider theme={theme}>
        <BasketContextProvider>
          <MemoryRouter initialEntries={['/products/1']}>
            {element}
          </MemoryRouter>
        </BasketContextProvider>
      </ThemeProvider>
    </MockedProvider>
  );
};

const waitForPage = async () => {
  await waitFor(() => screen.getByText('Loading product...'));

  jest.runAllTimers();

  return waitFor(() => screen.getByText(productFixture.name));
};

describe('Components.App', () => {
  it('should render as expected, without crashing', async () => {
    const { asFragment } = render(withWrapper(<App />));

    await waitForPage();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be able to increase and decrease product quantity', async () => {
    const { getByLabelText, getByRole } = render(withWrapper(<App />));

    await waitForPage();

    const increaseButton = getByLabelText('Increase Quantity');
    const decreaseButton = getByLabelText('Decrease Quantity');
    const quantityTextbox = getByRole('textbox');

    expect(quantityTextbox).toHaveTextContent('1');
    expect(getByLabelText('Quantity')).toBeInTheDocument();

    fireEvent.click(increaseButton);
    fireEvent.click(increaseButton);
    expect(quantityTextbox).toHaveTextContent('3');

    fireEvent.click(decreaseButton);
    expect(quantityTextbox).toHaveTextContent('2');
  });

  it('should be able to add items to the basket', async () => {
    const { queryByLabelText, getByLabelText, getByText } = render(
      withWrapper(<App />),
    );

    await waitForPage();

    const submitButton = getByText('Add to cart');

    expect(queryByLabelText('You have 1 products in your basket')).toBeNull();

    fireEvent.click(submitButton);

    expect(
      getByLabelText('You have 1 products in your basket'),
    ).toBeInTheDocument();
  });
});
