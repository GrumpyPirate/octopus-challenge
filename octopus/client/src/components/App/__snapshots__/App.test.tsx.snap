// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components.App should render as expected, without crashing 1`] = `
<DocumentFragment>
  .c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.75rem 1rem;
  background-color: #070c1f;
  z-index: 1;
}

.c1 {
  display: block;
  width: 12rem;
  height: 1.75rem;
}

.c2 {
  position: relative;
}

.c3 {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-appearance: none;
  touch-action: manipulation;
}

.c3::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c3 img {
  display: block;
  width: 1.875rem;
  height: 1.875rem;
}

.c4 {
  display: inline-block;
  padding: 0.125rem 0.375rem;
  background-color: #f60068;
  color: #fff;
  font-size: 0.75rem;
  font-weight: 700;
  border-radius: 0.5rem;
  position: absolute;
  top: 100%;
  right: 100%;
  -webkit-transform: translate(calc(50% + 0.375rem),calc(-50% - 0.375rem));
  -ms-transform: translate(calc(50% + 0.375rem),calc(-50% - 0.375rem));
  transform: translate(calc(50% + 0.375rem),calc(-50% - 0.375rem));
}

@media (min-width:48em) {

}

@media (min-width:64em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

<header
    class="c0"
  >
    <a
      href="/"
    >
      <img
        alt="Octopus Energy"
        class="c1"
        height="28"
        src="svg"
        width="192"
      />
    </a>
    <div
      class="c2"
    >
      <button
        aria-label="Your basket"
        class="c3"
        type="button"
      >
        <img
          alt=""
          src="svg"
        />
      </button>
      <span
        aria-label="You have 0 products in your basket"
        class="c4"
      >
        <span
          aria-hidden="true"
        >
          0
        </span>
      </span>
    </div>
  </header>
  .c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  font-size: 2rem;
  line-height: 2.5rem;
  font-weight: 700;
  padding: 0 0.5rem;
}

.c13 {
  font-size: 0.875rem;
  color: #5280d6;
  text-transform: uppercase;
  margin-bottom: 0.25rem;
}

.c15 {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-appearance: none;
  touch-action: manipulation;
  font-size: 1.5rem;
  line-height: 1;
  width: 2.5rem;
  height: 2.5rem;
  border-radius: 0.25rem;
  background-color: #5280d6;
}

.c15::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c15[disabled] {
  background-color: #151e33;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c7 {
  padding: 0.625rem 0 1.5rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1.25rem;
}

.c9 {
  font-size: 2.5rem;
  font-weight: 700;
  line-height: 1;
  white-space: nowrap;
  margin: 0;
}

.c10 {
  display: inline-block;
  vertical-align: top;
}

.c11 {
  position: relative;
  top: 0.0625em;
  display: inline-block;
  vertical-align: top;
  font-size: 0.75em;
}

.c17 {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-appearance: none;
  touch-action: manipulation;
  display: block;
  width: 100%;
  padding: 0.75rem 1.5rem;
  font-size: 1.5rem;
  font-weight: 700;
  text-align: center;
  border-radius: 0.25rem;
  background-color: #f60068;
  position: relative;
}

.c17::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c17::after {
  content: '';
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  height: 50%;
  pointer-events: none;
  background-image: linear-gradient( to bottom, rgba(255,255,255,0.5), rgba(255,255,255,0.25) );
  border-top-right-radius: inherit;
  border-top-left-radius: inherit;
}

.c17:active {
  background-color: #b6004d;
}

.c1 {
  position: relative;
  margin: 0 2.375rem;
  height: 24.375rem;
}

.c1 img {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: block;
  width: 100%;
  height: 100%;
  object-fit: contain;
  object-position: center;
}

.c3 {
  padding: 3rem 2.375rem 1.5rem;
  background-color: #02122d;
}

.c6 {
  padding: 0 2.375rem;
}

.c2 {
  margin-top: -2rem;
}

.c4 {
  font-size: 2.25rem;
  font-weight: 700;
  line-height: 1.15;
  margin: 0 0 1rem;
}

.c5 {
  color: #5280d6;
  font-size: 1.125rem;
  font-weight: 400;
  margin: 0;
}

@media (min-width:48em) {
  .c0 {
    display: grid;
    grid-template-columns: 16rem 1fr;
    grid-gap: 2.375rem;
    padding: 0 2.375rem;
  }
}

@media (min-width:64em) {
  .c0 {
    grid-template-columns: 20rem 1fr;
  }
}

@media (min-width:48em) {
  .c1 {
    margin: 0;
  }

  .c1 img {
    object-position: top center;
  }
}

@media (min-width:48em) {
  .c3 {
    background: none;
    padding-top: 0;
    padding-right: 0;
    padding-left: 0;
  }
}

@media (min-width:48em) {
  .c6 {
    padding: 0;
    max-width: 25rem;
  }
}

@media (min-width:48em) {
  .c2 {
    margin-top: 0;
  }
}

<header
    class="c0"
  >
    <figure
      class="c1"
    >
      <img
        alt=""
        src="https://i.ibb.co/2nzwxnQ/bulb.png"
      />
    </figure>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4"
        >
          Energy saving light bulb
        </h1>
        <h2
          class="c5"
        >
          25W // Packet of 4
        </h2>
      </div>
      <div
        class="c6"
      >
        <form
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c9"
            >
              <span
                class="c10"
              >
                12
              </span>
              <span
                class="c11"
              >
                .99
              </span>
            </h3>
            <label
              class="c12"
              for="product-quantity"
            >
              <div
                aria-label="Quantity"
                class="c13"
                id="product-quantity-label"
              >
                <span
                  aria-hidden="true"
                >
                  QTY
                </span>
              </div>
              <div
                class="c14"
              >
                <button
                  aria-label="Decrease Quantity"
                  class="c15"
                  disabled=""
                  type="button"
                >
                  -
                </button>
                <span
                  aria-labelledby="product-quantity-label"
                  aria-readonly="true"
                  class="c16"
                  role="textbox"
                >
                  1
                </span>
                <button
                  aria-label="Increase Quantity"
                  class="c15"
                  type="button"
                >
                  +
                </button>
              </div>
            </label>
          </div>
          <button
            class="c17"
            type="submit"
          >
            Add to cart
          </button>
        </form>
      </div>
    </div>
  </header>
  .c1 {
  width: 100%;
  max-width: 90rem;
  margin: 0 auto;
  padding: 0 2.375rem;
}

.c0 {
  padding: 2rem 0 1.75rem;
  background-color: #02122d;
}

.c2 {
  font-size: 1.75rem;
  font-weight: 700;
  margin: 0 0 1rem;
}

.c3 {
  margin: 0;
}

@media (min-width:48em) {

}

@media (min-width:64em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Description
      </h3>
      <p
        class="c3"
      >
        Available in 7 watts, 9 watts, 11 watts Spiral Light bulb in B22, bulb switches on instantly, no wait around warm start and flicker free features make for a great all purpose bulb
      </p>
    </div>
  </section>
  .c1 {
  width: 100%;
  max-width: 90rem;
  margin: 0 auto;
  padding: 0 2.375rem;
}

.c0 {
  padding: 2rem 0 1.75rem;
}

.c2 {
  font-size: 1.75rem;
  font-weight: 700;
  margin: 0 0 1rem;
}

.c3 {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 11rem 1fr;
  margin: 0;
}

.c4 {
  margin: 0;
}

@media (min-width:48em) {

}

@media (min-width:64em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Specifications
      </h3>
      <dl
        class="c3"
      >
        <dt>
          Brand
        </dt>
        <dd
          class="c4"
        >
          Philips
        </dd>
        <dt>
          Item weight
        </dt>
        <dd
          class="c4"
        >
          77
        </dd>
        <dt>
          Dimensions
        </dt>
        <dd
          class="c4"
        >
          6.2x12.6x6.2
        </dd>
        <dt>
          Item model number
        </dt>
        <dd
          class="c4"
        >
          E27 ES
        </dd>
        <dt>
          Colour
        </dt>
        <dd
          class="c4"
        >
          Cool daylight
        </dd>
      </dl>
    </div>
  </section>
  .c1 {
  width: 100%;
  max-width: 90rem;
  margin: 0 auto;
  padding: 0 2.375rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

.c0 {
  padding: 1.25rem 0 1.5rem;
  color: #7c90b4;
  font-size: 0.75rem;
  background-color: #02122d;
}

.c2 {
  margin: 0;
}

@media (min-width:48em) {

}

@media (min-width:64em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

@media (min-width:48em) {

}

<footer
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
      >
        Octopus Energy Ltd is a company registered in England and Wales.
        <br />
        Registered number: 09265424. Registered oftice: 35 Holborn London. ECIN 2HT. Trading office: 20-24 Broadwick Street London, WiF 8HT
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
